File: src/main/java/pro/sky/java/weatherhistoryapp/domain/HistoryRecord.java
package pro.sky.java.weatherhistoryapp.domain;

import lombok.*;
import lombok.experimental.Accessors;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Table
@Getter
@Setter
@ToString
@Accessors(chain = true)
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class HistoryRecord {
    @Id
    private Integer id;
    private String cityName;
    @CreatedDate
    private LocalDateTime createdAt;
    private String result;
}

File: src/main/kotlin/pro/sky/java/weatherhistoryapp/domain/HistoryRecord.kt
package pro.sky.java.weatherhistoryapp.domain

import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table
class HistoryRecord {
    @Id
    var id: Int? = null
    var cityName: String? = null

    @CreatedDate
    var createdAt: LocalDateTime? = null
    var result: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is HistoryRecord) return false

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id ?: 0
    }

    override fun toString(): String {
        return "HistoryRecord(id=$id, cityName=$cityName, createdAt=$createdAt, result=$result)"
    }
}
